woocommerce/woocommerce
# Package Release: handle initial release case

### All Submissions:






### Changes proposed in this Pull Request:



Closes https://github.com/woocommerce/woocommerce/issues/36632

This pull request alleviates difficulty in preparing a package for its initial release. There was problems of a changelog either not existing or being incorrectly formatted. This PR makes the following changes
* Update instructions for adding a new package to include a changelog file
* Adds a flag to indicate an initial release so the `1.0.0` version number can be used.
* Updates relevant documentation



### How to test the changes in this Pull Request:




`@woocommerce/product-editor` is a newly added package that has not been released. We will use the Prepare tool's CLI to test changes.

1. Try to prepare the package without the flag and see a useful error result.
```
./tools/package-release/bin/dev prepare @woocommerce/product-editor
```
2. Try to prepare the package along with another, this time using the flag. Make sure a useful error results
```
./tools/package-release/bin/dev prepare @woocommerce/product-editor,@woocommerce/number --initialRelease
```
3. Now prepare the package for release with the correct flag.
```
./tools/package-release/bin/dev prepare @woocommerce/product-editor --initialRelease
```
The resulting changes should be the compilation of changelog entries into the `packages/js/product-editor/changelog.md` file, which should be correctly configured. The version number in `package.json` should also be bumped.

4. Abandon all changes from the previous step and make sure the prepare script works on previously released packages.
```
./tools/package-release/bin/dev prepare @woocommerce/components
```
The changelog will be compiled and the version bumped in `package.json`.



### Other information:




### FOR PR REVIEWER ONLY:
